### ISSUE TITLE
bug report: nvda don't detect context menu close in javafx textArea control

### ISSUE THREAD

**Issue by ilyapashuk** [nvaccess/nvda#12747]
### Steps to reproduce:
start any javafx application with textArea control
open up context menu on this control
activate any item, for example selectAll item

### Actual behavior:
after press of Enter nvda speaks nothing and thinks that focus is remaining on the menu item.

### Expected behavior:
nvda should return to the editing field like in normal windows applications

### System configuration
nvda 2021.1 installed version

#### Windows version:
windows 10 1803 ltsc

#### Name and version of other software in use when reproducing the issue:
java 16 openjdk, javafx sdk 16

**Comment by feerrenrut** [comment]
@ilyapashuk Can you give the name of an example application, or better provide sample code?
If NVDA developers aren't able to reproduce the issue, it is unlikely to be investigated further.

After pressing enter, I understand that NVDA does not announce any change, but I want to know what the current system focus is.

- Please restart NVDA with debug logging enabled
- Reproduce the issue
- After pressing enter, press `NVDA+F1` to print the developer info to the log.
- Attach the log to this issue.

[There is more information about collecting logs on the NVDA wiki](https://github.com/nvaccess/nvda/wiki/LogFilesAndCrashDumps)

**Comment by ilyapashuk** [comment]
source code may be found hear:
https://drive.google.com/open?id=1k7beRfng-h005dZIGDPnb9iHBzoHLV4Z

I also packed it to the standalone package, which is available hear:
https://drive.google.com/open?id=1oQL-EdyIum9vNAbIgwLYOB-Fcz0FCPk2

**Comment by ilyapashuk** [comment]
and requested logs in the attachment, but I can't found anything usefull.
[nvda.log](https://github.com/nvaccess/nvda/files/7014445/nvda.log)

**Comment by feerrenrut** [comment]
Thanks for providing that.

The devInfo shows that the navigator object is not changed away from the menuItem.
First you'll have to confirm this not a bug in the java sample, it may be easier to contact the JavaFx developers to get help with debugging this.
- Are you confident that the menu does close?
- Where is the focus after pressing enter?
- Does a focus event get emitted by the application?

**Comment by ilyapashuk** [comment]
also if I use ctrl+left and ctrl+right to move by words, cursor position is not reported correctly.

**Comment by mwhapples** [comment]
Removing the Java access Bridge label as JavaFX does not use the Java Access Bridge.

**Comment by feerrenrut** [comment]
@ilyapashuk to help us prioritize this, can you let us know of any real-world use-case. I.E. an app you are trying to use, or build that is running into this problem?

We need to be confident that this is not a mistake in the sample, we don't have the familiarity with javaFX to be sure.

**Comment by ilyapashuk** [comment]
I seen this only in my sample, which is a self written application.

I hadn't yet used other javafx applications and don't know any real applications which are effected by this bug, but it doesn't mean that such apps are not exist.

**Comment by Adriani90** [comment]
@ilyapashuk the google drive link is not valid anymore. Are you still having this issue in NVDA 2024.4.1?

**Comment by Adriani90** [comment]
No updates upon request since almost one year, closing as abandoned. If you are still having this issue, pleaes provide updates as requested and we can reopen.
